src/imports.rs

```diff
@@ -89,10 +88,15 @@ impl<'a> FmtVisitor<'a> {
                 ast::PathListItem_::PathListMod{ .. } => None,
             }
         })).collect();
+        let list_str = if items.len() == 1 {
+            format!("{}", write_list(&items, &fmt))
+        } else {
+            format!("{{{}}}", write_list(&items, &fmt))
+        };
         if path_str.len() == 0 {
-            format!("{}use {{{}}};", vis, write_list(&items, &fmt))
+            format!("{}use {};", vis, list_str)
         } else {
-            format!("{}use {}::{{{}}};", vis, path_str, write_list(&items, &fmt))
+            format!("{}use {}::{};", vis, path_str, list_str)
```

Comment:
This solution is absolutely correct, however, I think it would be better, if you don't mind, to test earlier (near the start of the method) for the list only being length 1 and then skipping all the stuff about making a list and finding self, etc. (after all with only one item we can't insert line breaks or reorder). This would also make me feel better because `used_width` takes into account the `{}` which it doesn't need to and so might give incorrect results if the line would be exactly 100 chars wide.